app/helpers/hyrax/membership_helper.rb

```diff
@@ -41,7 +41,7 @@ def work_members_json(resource)
       Hyrax.custom_queries.find_child_works(resource: resource).map do |member|
         { id: member.id.to_s,
           label: member.title.first,
-          path: url_for(member) }
+          path: url_helpers.url_for([member, { only_path: true }]) }
```

Comment:
Does this work if you use `main_app` instead?
```suggestion
          path: main_app.url_for([member, { only_path: true }]) }
```
(See the end of [this section](https://guides.rubyonrails.org/engines.html#routes) in rails guide for engines)

Reply:
Thanks @cjcolvar! That is definitely what I was looking for, though I'm still unsure why this is required for Valkyrie resources. 🤷